File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.go
func numberOfArrays(differences []int, lower int, upper int) int {
	x := int64(0)
	maxX := int64(0)
	minX := int64(0)

	for _, difference := range differences {
		x += int64(difference)
		maxX = max(maxX, x)
		minX = min(minX, x)
	}

	start := int64(lower) - minX
	end := int64(upper) - maxX

	if start > end {
		return 0
	}
	return int(end - start + 1)
}

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.cpp
class Solution {
   public:
    int numberOfArrays(vector<int>& differences, int lower, int upper) {
        int n = differences.size();
        long long x = 0;
        long long maxX = 0;
        long long minX = 0;
        for (const auto& difference : differences) {
            x += difference;
            maxX = max(maxX, x);
            minX = min(minX, x);
        }

        long long start = lower - minX;
        long long end = upper - maxX;

        return start > end ? 0 : end - start + 1;
    }
};